### PR TITLE
David session 2

### PR DESCRIPTION
**Pull request by abutbul**
@gairadzi thank you for the help along the way!

please see two comments in HomeWork1/cookbooks/my-first-cookbook/spec/default_spec.rb
I wasn't able to nail down the method, would appreciate guidance

also
TIL, if you open a pull request, write a comment, commit again and open a new pull request, content persists! awesome work github :+1:

### CODE REVIEW

File: HomeWork1/cookbooks/my-first-cookbook/recipes/default.rb [hunk 1]
```diff
@@ -6,3 +6,52 @@
 #
 # All rights reserved - Do Not Redistribute
 #
+node['mycookbook']['packages'].each do |pkg|
+  package pkg do
+    action :install
+    options '--force-yes'
+	not_if "dpkg -l | grep #{pkg} | grep ii"
```
Comment (gairadzi) [inline HomeWork1/cookbooks/my-first-cookbook/recipes/default.rb]: There is no need for the 'not_if' in the package resource (it handles it on it's own - if the package is already installed the resource will not be updated)

File: HomeWork1/cookbooks/my-first-cookbook/recipes/default.rb [hunk 2]
```diff
@@ -6,3 +6,52 @@
...
+  command 'pip install flask '
+  action :run
+	not_if { File.exist?("/usr/local/lib/python2.7/dist-packages/flask") }
+end
+
+cookbook_file node['mycookbook']['app']['db_creation_script'] do
+  action :create
+  owner 'root'
+  group 'root'
+  notifies :run, 'execute[CREATE USER]', :delayed
```
Comment (gairadzi) [inline HomeWork1/cookbooks/my-first-cookbook/recipes/default.rb]: why did you choose delayed on this one?

File: HomeWork1/cookbooks/my-first-cookbook/templates/default/AAR_config.py.erb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+CONNECTION_ARGS = {"host":"localhost", "user":"aarapp", "passwd":"7ERwzg7E", "db":"AARdb"}
```
Comment (gairadzi) [inline HomeWork1/cookbooks/my-first-cookbook/templates/default/AAR_config.py.erb]: what about the attributes here?

File: HomeWork1/cookbooks/my-first-cookbook/recipes/default.rb (hunk 1 above)
Comment (abutbul) [inline HomeWork1/cookbooks/my-first-cookbook/recipes/default.rb]: thanks, will align!

File: HomeWork1/cookbooks/my-first-cookbook/recipes/default.rb (hunk 2 above)
Comment (abutbul) [inline HomeWork1/cookbooks/my-first-cookbook/recipes/default.rb]: can't recall a specific reason, will update this as well

File: HomeWork1/cookbooks/my-first-cookbook/templates/default/AAR_config.py.erb (hunk 3 above)
Comment (abutbul) [inline HomeWork1/cookbooks/my-first-cookbook/templates/default/AAR_config.py.erb]: I managed to forget this part, will fix and re-request pull
thank you for taking the time to review!

**Pull request closed without merging**